### PR TITLE
[web] Warn users when picking a deprecated renderer.

### PR DESCRIPTION
**Pull request by ditman**
Warn users when they pick a deprecated renderer for the web, and point them to the appropriate /go/ link.

## Issues

* Part of https://github.com/flutter/flutter/issues/145954
* Fixes https://github.com/flutter/flutter/issues/154879

## Pre-launch Checklist

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I read and followed the [Flutter Style Guide] and the [C++, Objective-C, Java style guides].
- [x] I listed at least one issue that this PR fixes in the description above.
- [x] I added new tests to check the change I am making or feature I am adding, or the PR is [test-exempt]. See [testing the engine] for instructions on writing and running engine tests.
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] I signed the [CLA].
- [x] All existing and new tests are passing.

If you need help, consider asking for advice on the #hackers-new channel on [Discord].

[Contributor Guide]: https://github.com/flutter/flutter/wiki/Tree-hygiene#overview
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[test-exempt]: https://github.com/flutter/flutter/wiki/Tree-hygiene#tests
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[C++, Objective-C, Java style guides]: https://github.com/flutter/engine/blob/main/CONTRIBUTING.md#style
[testing the engine]: https://github.com/flutter/flutter/wiki/Testing-the-engine
[CLA]: https://cla.developers.google.com/
[flutter/tests]: https://github.com/flutter/tests
[breaking change policy]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes
[Discord]: https://github.com/flutter/flutter/wiki/Chat

### CODE REVIEW

File: lib/web_ui/lib/src/engine/configuration.dart [hunk 1]
```diff
@@ -177,10 +177,10 @@ class FlutterConfiguration {
 
   /// Enable the Skia-based rendering backend.
   ///
-  /// Using flutter tools option "--web-render=canvaskit" would set the value to
+  /// Using flutter tools option "--web-renderer=canvaskit" would set the value to
```
Comment (kevmoo) [inline lib/web_ui/lib/src/engine/configuration.dart:180]: Good fix!

**Review by kevmoo — APPROVED** [review]

**Review by kevmoo — CHANGES_REQUESTED** [review]
HOLD PLEASE! Need to update the link!

File: lib/web_ui/lib/src/engine/renderer.dart [hunk 2]
```diff
@@ -38,6 +38,19 @@ abstract class Renderer {
         useCanvasKit = FlutterConfiguration.useSkia;
       }
 
+      // Warn users in development that anything other than canvaskit is deprecated.
+      assert(() {
+        if (!useCanvasKit) {
```
Comment (yjbanov) [inline lib/web_ui/lib/src/engine/renderer.dart:43]: Will this trigger if you pick `auto` but are debugging in desktop Chrome (I imagine it's the most common case)? I think we want `if (['html', 'auto'].contains(configuration.requestedRendererType))`.

File: lib/web_ui/lib/src/engine/renderer.dart [hunk 3]
```diff
@@ -38,6 +38,19 @@ abstract class Renderer {
...
+          // The user requested 'html' or 'auto' either in the command-line or JS.
+          final String requested =
+            configuration.requestedRendererType ??
+            (FlutterConfiguration.flutterWebAutoDetect ? 'auto' : 'html');
+          printWarning(
+            'The HTML Renderer is being deprecated. Stop using the "$requested" renderer mode.'
+            '\nSee: https://docs.flutter.dev/go/web-html-renderer-deprecation');
```
Comment (kevmoo) [inline lib/web_ui/lib/src/engine/renderer.dart:50]:
```suggestion
            '\nSee: https://docs.flutter.dev/to/web-html-renderer-deprecation');
```

See https://github.com/flutter/website/pull/11238

**Review by kevmoo — CHANGES_REQUESTED** [review]

File: lib/web_ui/lib/src/engine/renderer.dart (hunk 2 above)
Comment (ditman) [reply lib/web_ui/lib/src/engine/renderer.dart:43]: In the current branch of the if, skwasm has already been chosen, and when we hit this we know that the HTML renderer is being selected, we're trying to reconstruct why.

I think the only corner case that this won't pick up at runtime is if the user has set `--web-renderer=auto` but then used `renderer: canvaskit` in their [load() call](https://docs.flutter.dev/platform-integration/web/initialization#the-_flutter-loader-load-api).

File: lib/web_ui/lib/src/engine/renderer.dart (hunk 3 above)
Comment (ditman) [reply lib/web_ui/lib/src/engine/renderer.dart:50]: Roger that!

Comment (ditman) [reply lib/web_ui/lib/src/engine/renderer.dart]: Linked to:

https://docs.flutter.dev/to/web-html-renderer-deprecation

**Review by kevmoo — APPROVED** [review]

**Comment by ditman** [comment]
(The tests failings seem to be related to impeller, unrelated to this change. Hit retry just in case they were flakes.)

File: lib/web_ui/lib/src/engine/renderer.dart (hunk 2 above)
Comment (yjbanov) [reply lib/web_ui/lib/src/engine/renderer.dart:43]: What is the value of `useCanvasKit` when you do `--web-renderer=auto` and test in Chrome?

**Review by eyebrowsoffire — APPROVED** [review]
LGTM

**Review by yjbanov — APPROVED** [review]

File: lib/web_ui/lib/src/engine/renderer.dart (hunk 2 above)
Comment (ditman) [reply lib/web_ui/lib/src/engine/renderer.dart:43]: > What is the value of useCanvasKit when you do --web-renderer=auto and test in desktop Chrome?

`true` (as expected, no?)

Comment (mdebbar) [reply lib/web_ui/lib/src/engine/renderer.dart:43]: I think we could've done:
```dart
if (FlutterConfiguration.flutterWebAutoDetect || !useCanvasKit) {
```
to capture all cases we want to warn against.

**Pull request merged**